File: src/test/java/de/pflugradts/pwman3/application/PasswordStoreAdapterPortFaker.java
package de.pflugradts.pwman3.application;

import de.pflugradts.pwman3.domain.model.password.PasswordEntry;
import de.pflugradts.pwman3.domain.service.password.storage.PasswordStoreAdapterPort;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import static org.mockito.BDDMockito.given;
import static org.mockito.Mockito.mock;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class PasswordStoreAdapterPortFaker {

    private PasswordStoreAdapterPort passwordStoreAdapterPort = mock(PasswordStoreAdapterPort.class);
    private List<PasswordEntry> passwordEntries = new ArrayList<>();

    public static PasswordStoreAdapterPortFaker faker() {
        return new PasswordStoreAdapterPortFaker();
    }

    public PasswordStoreAdapterPortFaker forInstance(final PasswordStoreAdapterPort passwordStoreAdapterPort) {
        this.passwordStoreAdapterPort = passwordStoreAdapterPort;
        return this;
    }

    public PasswordStoreAdapterPortFaker withEmptyPasswordEntriesList() {
        passwordEntries.clear();
        return this;
    }

    public PasswordStoreAdapterPortFaker withThesePasswordEntries(final PasswordEntry... passwordEntries) {
        this.passwordEntries.clear();
        Collections.addAll(this.passwordEntries, passwordEntries);
        return this;
    }

    public PasswordStoreAdapterPort fake() {
        given(passwordStoreAdapterPort.restore()).willReturn(() -> passwordEntries.stream());
        return passwordStoreAdapterPort;
    }

}

File: src/test/kotlin/de/pflugradts/passbird/domain/service/password/storage/PasswordStoreAdapterPortFaker.kt
package de.pflugradts.passbird.domain.service.password.storage

import de.pflugradts.passbird.domain.model.egg.Egg
import io.mockk.every
import io.mockk.mockk
import java.util.function.Supplier

fun fakePasswordStoreAdapterPort(withEggs: List<Egg> = emptyList()): PasswordStoreAdapterPort {
    val instance = mockk<PasswordStoreAdapterPort>()
    every { instance.restore() } returns Supplier { withEggs.stream() }
    every { instance.sync(any()) } returns Unit
    return instance
}
